power_grid_model_c/power_grid_model/include/power_grid_model/main_core/input.hpp

```diff
@@ -136,7 +136,7 @@ inline void add_component(MainModelState<ComponentContainer>& state, ForwardIter
                     throw InvalidRegulatedObject(input.regulated_object, Component::name);
                 }
             }();
-
+            // (TODO: jguo) This validation still needs to be updated
```

Comment:
@nitbharambe Shall we quickly go over this part and check whether it is actually needed?

Reply:
How does it look now? ;)

Reply:
We still need the InvalidRegulatedObject check. ie. when component is not a trafo or a trafo3w.

Reply:
Addressed.